fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java

```diff
@@ -46,6 +48,9 @@ public <T> boolean canSerialize(BusinessEvent<T> event) {
     protected <T> ByteBufferSerializable toAvroDTO(BusinessEvent<T> rawEvent) {
         LoanBusinessEvent event = (LoanBusinessEvent) rawEvent;
         LoanAccountData data = service.retrieveOne(event.get().getId());
+        if (event instanceof LoanApprovedBusinessEvent || event instanceof LoanDisbursalBusinessEvent) {
```

Comment:
This is not gonna fly. Either let's create a specific Serializer for these 2 events and mark it as priority. An example for that is `LoanChargeDeletedBusinessEventSerializer`

Or, rather fetch all the data for every LoanBusinessEvent (including the repayment schedule data).

Reply:
Done, It was set for every LoanBusinessEvent